includes/annotationcommand_adapter.inc

```diff
@@ -329,19 +354,100 @@ function annotationcommand_adapter_get_commands_for_commandhandler($commandhandl
         $command['options'][$option] = $fn($command['options'][$option], $commandinfo, $default);
       }
     }
+    $command['annotations'] = $commandinfo->getAnnotations();
     // If the command has a '@return' annotation, then
     // remember information we will need to use the output formatter.
     $returnType = $commandinfo->getReturnType();
     if (isset($returnType)) {
-      $command['annotations'] = $commandinfo->getAnnotations();
       $command['return-type'] = $returnType;
     }
     $command += drush_command_defaults($command_name, $commandfile, $commandfile_path);
     $commands[$command_name] = $command;
   }
+  $cache[$commandfile_path] = $commands;
   return $commands;
 }
 
+function annotationcommand_adapter_add_hook_options(&$command)
+{
+  // Get options added by hooks.  We postpone doing this until the
+  // last minute rather than doing it when processing commandfiles
+  // so that we do not need to worry about what order we process the
+  // commandfiles in -- we can load extensions late, and still have
+  // the extension hook a core command, or have an early-loaded global
+  // extension hook a late-loaded extension (e.g. attached to a module).
+  $names = annotationcommand_adapter_command_names($command);
+  $names[] = '*'; // we are missing annotations here; maybe we just don't support that? (TODO later, maybe)
+  $factory = annotationcommand_adapter_get_factory();
+  $extraOptions = $factory->hookManager()->getHookOptions($names);
+  foreach ($extraOptions as $commandinfo) {
+    if (!isset($command['consolidation-option-defaults'])) {
+      $command['consolidation-option-defaults'] = array();
+    }
+    $command['consolidation-option-defaults'] += $commandinfo->options()->getValues();
+    foreach ($commandinfo->options()->getValues() as $option => $default) {
+      $description = $commandinfo->options()->getDescription($option);
+      $command['options'][$option] = ['description' => $description];
+      if (!empty($default)) {
+        $command['options'][$option]['example-value'] = $default;
+      }
+      $fn = 'annotationcommand_adapter_alter_option_description_' . $option;
+      if (function_exists($fn)) {
+        $command['options'][$option] = $fn($command['options'][$option], $commandinfo, $default);
+      }
+    }
+  }
+}
+
+function annotationcommand_adapter_command_names($command)
+{
+  $names = array_merge(
+    [$command['name']],
+    $command['aliases']
+  );
+  if (!empty($command['annotated-command-callback'])) {
+    $commandHandler = $command['annotated-command-callback'][0];
+    $reflectionClass = new \ReflectionClass($commandHandler);
+    $commandFileClass = $reflectionClass->getName();
+    $names[] = $commandFileClass;
+  }
+  return $names;
+}
+
+function annotationcommand_adapter_call_hook_pre_validate($names, CommandData $commandData)
```

Comment:
I implemented all of these.  Let me check and see if I forgot to push that commit.